lib/dry/web/roda/generators/abstract_project.rb

```diff
@@ -0,0 +1,52 @@
+require "dry/web/roda/generate"
+
+module Dry
+  module Web
+    module Roda
+      module Generators
+        class AbstractProject
+          attr_reader :target_dir, :options
+
+          def initialize(target_dir, options = {})
+            @target_dir = target_dir
+            @options = options
+            @generator = Generate.new(destination)
+            @templates = []
```

Comment:
Perhaps we could do something like this right here in the initializer?

```ruby
@templates = []
map_templates
```

That way they're all generated up front and there isn't any danger of the templates array being accidentally over-populated if we ever happen to call `map_templates` more than once. Perhaps we could also call it `#populate_templates` and then make it private too.